lib/pure/unittest.nim

```diff
@@ -395,7 +443,7 @@ template test*(name, body) {.dirty.} =
 
   ensureInitialized()
 
-  if shouldRun(name):
+  if shouldRun(when declared(testSuiteName): testSuiteName else: "", name):
```

Comment:
Is this just for testing? I would remove it, makes the code more confusing.

Reply:
shouldRun needs access to testSuiteName, when it is declared, to do the filtering.